core/node/eigenda_proxy/src/eigenda_proxy_client.rs

```diff
@@ -0,0 +1,223 @@
+use std::{
+    fmt::Debug,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use rlp::decode;
+use tokio::{sync::Mutex, time::interval};
+use tonic::transport::{Channel, ClientTlsConfig};
+use zksync_config::configs::da_client::eigen_da::DisperserConfig;
+
+use crate::{
+    blob_info::BlobInfo,
+    disperser::{self, disperser_client::DisperserClient, BlobStatusRequest, DisperseBlobRequest},
+};
+
+pub struct EigenDAProxyClient {
```

Comment:
I don't think it should be named EigenDAProxyClient since there is no proxy here.